main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -70,118 +60,61 @@
+      for (GtfsRoute route : routes) {
+        noticeContainer.addValidationNotice(new SameStopAndRouteUrlNotice(stop, route));
       }
     }
-    Map<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
-    Map<String, GtfsStop> stopsByUrlMap = stopsByUrlMap(stopTable);
-
-    Maps.filterEntries(
-            stopsByUrlMap, entry -> routesByUrlMap.get(entry.getValue().stopUrl()) != null)
-        .values()
-        .forEach(
-            stopWithDuplicateUrl -> {
-              noticeContainer.addValidationNotice(
-                  new SameStopAndRouteUrlNotice(
-                      stopWithDuplicateUrl.csvRowNumber(),
-                      stopWithDuplicateUrl.stopId(),
-                      stopWithDuplicateUrl.stopUrl(),
-                      routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).routeId(),
-                      routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).csvRowNumber()));
-            });
-  }
-
-  /**
-   * Maps {@code GtfsAgency}s by there URLs if provided.
-   *
-   * @param stopTable the {@code GtfsStopTableContainer} to extract {@code GtfsStop} from
-   * @return routes from {@code GtfsStopTableContainer}s mapped by there {@code routes.route_url}
-   *     (in lower case) if provided.
-   */
-  private Map<String, GtfsStop> stopsByUrlMap(GtfsStopTableContainer stopTable) {
-    Map<String, GtfsStop> stopsByUrl = new HashMap<>();
-    stopTable
-        .getEntities()
-        .forEach(
-            stop -> {
-              if (stop.hasStopUrl()) {
-                if (stopsByUrl.get(stop.stopUrl()) == null) {
-                  stopsByUrl.put(stop.stopUrl().toLowerCase(), stop);
-                }
-              }
-            });
-    return stopsByUrl;
   }
 
   /**
    * Maps {@code GtfsRoute}s by their URLs if provided.
    *
    * @param routeTable the {@code GtfsRouteTableContainer} to extract {@code GtfsRoute} from
-   * @return routes from {@code GtfsRouteTableContainer}s mapped by there {@code routes.route_url}
+   * @return routes from {@code GtfsRouteTableContainer}s mapped by their {@code routes.route_url}
    *     (in lower case) if provided.
    */
-  private Map<String, GtfsRoute> routesByUrlMap(GtfsRouteTableContainer routeTable) {
-    Map<String, GtfsRoute> routesByUrl = new HashMap<>();
-    routeTable
-        .getEntities()
-        .forEach(
-            route -> {
-              if (route.hasRouteUrl()) {
-                if (routesByUrl.get(route.routeUrl()) == null) {
-                  routesByUrl.put(route.routeUrl().toLowerCase(), route);
-                }
-              }
-            });
+  private ListMultimap<String, GtfsRoute> routesByUrlMap(GtfsRouteTableContainer routeTable) {
+    ListMultimap<String, GtfsRoute> routesByUrl = ArrayListMultimap.create();
+    for (GtfsRoute route : routeTable.getEntities()) {
+      if (route.hasRouteUrl()) {
+        routesByUrl.put(Ascii.toLowerCase(route.routeUrl()), route);
```

Comment:
To tell the truth, only domain name is case-insensitive. The rest of a URL is case-sensitive (see, e.g., https://www.ibm.com/docs/en/sva/9.0?topic=configuration-support-urls-as-not-case-sensitive)

I think that either approach is OK for the validator:
* fully case-insensitive (as you did)
* or fully case-sensitive.

Reply:
Thanks for the docs. I'll keep this in mind in case we need to treat the urls as fully sensitive.